src/tiled/mapeditor.cpp

```diff
@@ -314,6 +316,10 @@ void MapEditor::addDocument(Document *document)
         if (Layer *layer = layerAtGlobalIndex(mapDocument->map(), layerIndex))
             mapDocument->setCurrentLayer(layer);
     }
+
+    // Update all templates instances in this MapDocument when a template is edited
+    // Maybe it can be optimized
```

Comment:
Hmm, I had the feeling that live updating might be slow. I tried it anyway but found that the object itself wasn't updated until the undo index is changed, i.e. the object is not saved mid-edit, so calling sync for the template instances didn't do anything. So it might require some other way of syncing, I think we can leave it for now.